presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -43,6 +44,14 @@ private DependencyExtractor() {}
         return uniqueSymbols.build();
     }
 
+    public static Set<Symbol> extractUniqueNonRecursive(PlanNode node)
```

Comment:
Ok, maybe we need something like this for this specific use case. However, the whole abstraction of DependencyExtractor seems a bit broken -- maybe as a result of way the current plan structure representation works. 

The notion of getting dependencies recursively seems odd, since "dependency" just means the "inputs to a given node or subplan" (i.e., the dependencies of a subplan would be the inputs to the leaves in the tree an any input from the enclosing context provided via correlated symbols).

Hence my suggestion that we might want to take a look at whether what we need is to fix extractUnique, instead.